### PR TITLE
fixed the issues with web3 apis to support ethereum clients

### PR DESCRIPTION
**Pull request by dushyantb**

### CODE REVIEW

File: src/Lachain.Core/RPC/HTTP/Web3/AccountServiceWeb3.cs [hunk 1]
```diff
@@ -44,10 +45,10 @@ public string GetBalance(string address, string tag)
         }
 
         [JsonRpcMethod("eth_getTransactionCount")]
-        public ulong GetTransactionCount(string from, string blockId)
+        public HexBigInteger GetTransactionCount(string from, string blockId)
```
Comment (sgladkov) [inline src/Lachain.Core/RPC/HTTP/Web3/AccountServiceWeb3.cs:48]: We have our own format utils,  I don't think it is a good idea to use other project formats in parallel, Remove string or JObject instead and format it as required

File: src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs [hunk 2]
```diff
@@ -142,28 +142,30 @@ public string VerifyRawTransaction(string rawTx)
         }
 
         [JsonRpcMethod("eth_getTransactionByBlockHashAndIndex")]
-        public JObject? GetTransactionByBlockHashAndIndex(string blockHash, ulong index)
+        public JObject? GetTransactionByBlockHashAndIndex(string blockHash, string index)
         {
+            ulong txIndex = index.HexToUlong();
```
Comment (sgladkov) [inline src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs:147]: Are you sure index should be in hex format only?

File: src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs [hunk 3]
```diff
@@ -142,28 +142,30 @@ public string VerifyRawTransaction(string rawTx)
...
+            var txHash = block.TransactionHashes[(int)txIndex];
             var receipt = _stateManager.LastApprovedSnapshot.Transactions.GetTransactionByHash(txHash);
             return receipt is null ? null : Web3DataFormatUtils.Web3Transaction(receipt!, block?.Hash, receipt.Block);
         }
 
         [JsonRpcMethod("eth_getTransactionByBlockNumberAndIndex")]
-        public JObject? GetTransactionByBlockNumberAndIndex(string blockTag, ulong index)
+        public JObject? GetTransactionByBlockNumberAndIndex(string blockTag, string index)
         {
+            ulong txIndex = index.HexToUlong();
```
Comment (sgladkov) [inline src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs:161]: The same,  are you sure index is always hex string?

**Review by sgladkov — CHANGES_REQUESTED** [review]

File: src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs (hunk 2 above)
Comment (dushyantb) [reply src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs:147]: https://infura.io/docs/ethereum/json-rpc/eth-getTransactionCount here is the details, where it is accepting the hex string as a input

File: src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs (hunk 3 above)
Comment (dushyantb) [reply src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs:161]: https://infura.io/docs/ethereum/json-rpc/eth-getTransactionCount here is the details, where it is accepting the hex string as a input

File: src/Lachain.Core/RPC/HTTP/Web3/AccountServiceWeb3.cs (hunk 1 above)
Comment (dushyantb) [reply src/Lachain.Core/RPC/HTTP/Web3/AccountServiceWeb3.cs:48]: fixed the issues in the latest commit

**Review by sgladkov — APPROVED** [review]

**Pull request merged**